pkg/operator/starter.go

```diff
@@ -326,3 +334,58 @@ func apiServicesReferences() []configv1.ObjectReference {
 	}
 	return ret
 }
+
+// ensureDaemonSetCleanup removes the daemonset used to manage apiserver pods
+// in releases prior to 4.5. The daemonset is removed once the deployment now
+// managing apiserver pods reports at least one pod available.
+func ensureDaemonSetCleanup(kubeClient *kubernetes.Clientset, stopCh <-chan struct{}) {
+	// daemonset and deployment both use the same name
+	resourceName := "apiserver"
+
+	dsClient := kubeClient.AppsV1().DaemonSets(operatorclient.TargetNamespace)
+	deployClient := kubeClient.AppsV1().Deployments(operatorclient.TargetNamespace)
+
+	// It's safe not to check the error returned by the poll function so long
+	// as conditionFunc never returns an error. The poll function will only
+	// exit on success or if the stop channel halts execution.
+	go wait.PollImmediateUntil(time.Minute, func() (bool, error) {
```

Comment:
`UntilWithContext` please.  This should run forever.  I'm ok doing it on two cadences if you prefer (fast until we delete, slow after that), but this will run every time the pod restarts, which is just a controller on an unpredictable resync cadence.

Reply:
Done. I didn't bother with 2 cadences, can address that if you feel strongly.